docs/advanced-topics/docker-alternatives.rst

```diff
@@ -0,0 +1,130 @@
+Docker Alternatives
+===================
+
+In some situations using Docker may be impractical because it requires all users to have root access.
+Several alternatives have been developed to make it possible to run rootless containers, including
+`Singularity <https://sylabs.io/docs/>`_,
+`rootless Docker <https://engineering.docker.com/2019/02/experimenting-with-rootless-docker/>`_,
+and `udocker <https://indigo-dc.gitbook.io/udocker/>`_.
+While Dockstore uses Docker by default, if necessary it may be possible to run your workflows with one
+of these alternatives. Because the call to Docker or an alternative is made by the workflow runner, usually cwltoool
+or Cromwell, and not Dockstore directly, the difficulty of configuring a Docker alternative depends on the workflow
+type. Some Dockstore entries will run seamlessly without Docker, and some may be entirely incompatible in a rootless
+environment.
+
+Singularity
+-----------
+
+Singularity is perhaps the most well-supported Docker alternative. Singularity can pull Docker images and build them
+into its own image format (.sif), but not all Docker features are compatible. For instance, dockerfile ``USER``
```

Comment:
Can also be a note or hint box, something like 
"A best practice when using Docker for workflows is not to rely on a specific user, this is doubly true for Singularity where it is not just best practice but necessary"

~Can link to something appropriate from https://github.com/common-workflow-language/common-workflow-language/issues/347~

Actually, I see you link to https://sylabs.io/guides/3.4/user-guide/singularity_and_docker.html#best-practices later, that works even better